drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -610,49 +610,55 @@
-          decapsulating node is also the Validator, for obvious performance
-          reasons.</t>
-
-          <t>The Validator recreates the signature over IOAM-Data-Fields collected
-          and checks the integrity against the Signature field. In order to
-          recompute the signature, the Validator iteratively follows the same
-          procedure as for the encapsulating, transit and decapsulating nodes,
-          in that order. It is trivial in some cases (e.g., POT Type-0 or E2E),
-          where only the encapsulating node generates a signature. For other
-          cases where transit nodes also generate a signature (e.g., Trace
-          Option-Types), node-ids MUST be recorded. Details on how the mapping
-          between node-ids and keys is implemented are outside the scope of this
-          document.</t>
+          sign(Signature || hash(IOAM-Data-Fields)). IOAM-Data-Fields modified
+          in-place by the transit node MUST be excluded from the signature
+          (e.g., the POT Cumulative field). The signature is stored in the Signature
+          field of the Integrity Protection subheader. Important note: if the
+          transit node does not insert IOAM-Data-Fields (e.g., it only modifies
+          IOAM-Data-Fields in-place, or does nothing), then the transit node
+          MUST NOT generate a signature and MUST NOT update the Signature field.</t>
+
+          <t>The decapsulating node (aka the Validator) is responsible for the
+          integrity verification of the IOAM-Data-Fields collected. For obvious
+          performance reasons and because it is the Validator, the decapsulating
+          node MUST NOT generate a signature based on IOAM-Data-Fields it has
+          inserted, if any, and therefore MUST NOT update the Signature field.
+          In order to validate the IOAM-Data-Fields integrity, the Validator
+          recomputes the signature by iteratively following the same procedure
+          as for the encapsulating and transit nodes, in that order, using their
+          respective keys (see <xref target="SymmetricMethod" /> or 
+          <xref target="AsymmetricMethod" /> depending on the approach, i.e., symmetric
+          or asymmetric). The recomputed signature is then compared to the
+          Signature field. It is trivial in some cases (e.g., with POT Type-0 or
+          E2E Option-Types), where only the encapsulating node generates a
+          signature, as specified by the method described in the current section.
+          Details on how the mapping between encapsulating nodes and keys is
+          implemented on the Validator are outside the scope of this document.
+          For other cases where transit nodes also generate a signature (e.g.,
+          with Trace Option-Types), node-ids or equivalents MUST be included in
+          IOAM-Data-Fields. Details on how the mapping between node-ids and keys
```

Comment:
Could we remove the earlier note in line 653 that I also left a comment on and simply use this more general statement here? IMHO this statement covers the earlier one.

Reply:
We could, but this is actually a distinct (easier) case where only the encapsulating node generates a signature. So, in that case, you wouldn't need for node-ids, which avoids issues of having both POT and a Trace (for instance) where you're not sure that the trace data fields will represent the same path as POT (or any other (future) Option-Type)... "do all nodes insert data? are they configured to do so? etc".

Reply:
We could, but this is actually a distinct (easier) case where only the encapsulating node generates a signature. So, in that case, you wouldn't need for node-ids. But I agree it would be more generic to go the way you suggest. Thoughts?

Reply:
Personally, I'm more for the "catch all" approach with a single sentence. These two sentences that were very similar in nature made me read the text multiple times. I get the nuance, but IMHO the "catch all" approach would cut it.

Reply:
It is indeed easier to read and understand. Let's do this :-)